src/vehicle.cpp

```diff
@@ -3815,8 +3857,88 @@ void vehicle::idle(bool on_map) {
 
     if( on_map ) {
         update_time();
-        if(scoop_on){
-            operate_scoop();
+    }
+}
+
+void vehicle::on_move(){
+    if(scoop_on){
+        operate_scoop();
+    }
+    if( planter_on ){
+        operate_planter();
+    }
+    if( plow_on ){
+        operate_plow();
+    }
+    if( reaper_on ){
+        operate_reaper();
+    }
+}
+void vehicle::operate_plow(){
+    for( const int plow_id : all_parts_with_feature( "PLOW" ) ){
+        const tripoint start_plow = global_pos3() + parts[plow_id].precalc[0];
+        if( g->m.has_flag("DIGGABLE", start_plow) ){
+            g->m.ter_set( start_plow, t_dirtmound );
+        } else {
+            const int speed = velocity;
+            const int v_damage = rng( 3, speed );
+            damage( plow_id, v_damage, DT_BASH, false );
+            sounds::sound( start_plow, v_damage, _("Clanggggg!") );
+        }
+    }
+}
+
+void vehicle::operate_reaper(){
+    const tripoint &veh_start = global_pos3();
+    for( const int reaper_id : all_parts_with_feature( "REAPER" ) ){
+        const tripoint start_reaper = veh_start + parts[reaper_id].precalc[0];
+        const int plant_produced =  rng( 1, parts[reaper_id].info().bonus );
+        const int seed_produced = rng(1, 3);
+        item tmp;
+        const tripoint &reaper_pos = start_reaper + parts[reaper_id].precalc[0];
```

Comment:
Beside the seemingly random usage of `const tripoint &` vs `const tripoint` (reference vs value, technically fine, but very confusing), what in Barry's name is that coordinate calculation supposed to be?

```C++
const tripoint start_reaper = veh_start + parts[reaper_id].precalc[0];
```
Ok, the absolute position of the reaper part in the map coordinate system.

```C++
const tripoint &reaper_pos = start_reaper + parts[reaper_id].precalc[0];
```
Huh? Why add `precalc[0]` again? This is btw. the only usage of `start_reaper`. The line is identical to
```C++
const tripoint &reaper_pos = veh_start + parts[reaper_id].precalc[0] + parts[reaper_id].precalc[0];
```